lib/util/content_steering_manager.js

```diff
@@ -0,0 +1,364 @@
+        return;
+      }
+      if (this.updateTimer_ != null) {
+        this.updateTimer_.stop();
+        this.updateTimer_ = null;
+      }
+      this.updateTimer_ = new shaka.util.Timer(() => {
+        this.requestInfo(uri);
+      });
+      this.updateTimer_.tickAfter(this.lastTTL_);
+    }
+  }
+
+  /** @private */
+  addQueryParams_(uri) {
+    if (!this.pathwayPriority_.length) {
+      return uri;
+    }
+    const finalUri = new goog.Uri(uri);
+    const currentPathwayID = this.lastPathwayUsed_ || this.pathwayPriority_[0];
+    const currentBandwidth =
+        Math.round(this.playerInterface_.getBandwidthEstimate());
+    const queryData = finalUri.getQueryData();
+    if (this.manifestType_ == shaka.media.ManifestParser.DASH) {
+      queryData.add('_DASH_pathway', currentPathwayID);
+      queryData.add('_DASH_throughput', String(currentBandwidth));
+    } else if (this.manifestType_ == shaka.media.ManifestParser.HLS) {
+      queryData.add('_HLS_pathway', currentPathwayID);
+      queryData.add('_HLS_throughput', String(currentBandwidth));
+    }
+    if (queryData.getCount()) {
+      finalUri.setQueryData(queryData);
+    }
+    return finalUri.toString();
+  }
+
+  /**
+   * @param {shaka.util.ContentSteeringManager.SteeringManifest} manifest
+   * @param {string} finalManifestUri
+   * @private
+   */
+  processManifest_(manifest, finalManifestUri) {
+    if (this.updateTimer_ != null) {
+      this.updateTimer_.stop();
+      this.updateTimer_ = null;
+    }
+    const uri = manifest['RELOAD-URI'] || finalManifestUri;
+    this.updateTimer_ = new shaka.util.Timer(() => {
+      this.requestInfo(uri);
+    });
+    if (manifest.TTL) {
+      this.lastTTL_ = 1;
+    }
+    this.updateTimer_.tickAfter(this.lastTTL_);
+    this.pathwayPriority_ = manifest['PATHWAY-PRIORITY'] || [];
+    this.pathwayClones_ = manifest['PATHWAY-CLONES'] || [];
+  }
+
+  /**
+   * Clear the previous locations added.
+   */
+  clearPreviousLocations() {
+    this.locations_.clear();
+  }
+
+  /**
+   * @param {string} streamId
+   * @param {string} pathwayId
+   * @param {string} uri
+   */
+  addLocation(streamId, pathwayId, uri) {
+    let streamLocations = this.locations_.get(streamId);
+    if (!streamLocations) {
+      streamLocations = new Map();
+    }
+    streamLocations.set(pathwayId, uri);
+    this.locations_.set(streamId, streamLocations);
+  }
+
+  /**
+   * @param {string} uri
+   */
+  banLocation(uri) {
+    const bannedUntil = Date.now() + 60000;
+    this.bannedLocations_.set(uri, bannedUntil);
+  }
+
+  /**
+   * Get the base locations ordered according the priority.
+   *
+   * @param {string} streamId
+   * @return {!Array.<string>}
+   */
+  getLocations(streamId) {
+    const streamLocations = this.locations_.get(streamId) || new Map();
+    let locationsMap = [];
+    for (const pathwayId of this.pathwayPriority_) {
+      const location = streamLocations.get(pathwayId);
+      if (location) {
+        locationsMap.push({
```

Comment:
This could just be:
`locationsMap.push({pathwayId, location});`

Reply:
Changed!